sdk/python/feast/feature_view.py

```diff
@@ -63,13 +66,21 @@ def __init__(
         name: str,
         entities: List[str],
         ttl: Optional[Union[Duration, timedelta]],
-        input: DataSource,
+        input: Optional[DataSource] = None,
         batch_source: Optional[DataSource] = None,
         stream_source: Optional[DataSource] = None,
         features: List[Feature] = None,
         tags: Optional[Dict[str, str]] = None,
         online: bool = True,
     ):
+        warnings.warn(
+            (
+                "The argument 'input' is being deprecated. Please use 'batch_source' "
+                "instead. Feast 0.12 and onwards will not support the argument 'input'."
```

Comment:
This should be feast 0.13 (since 0.12 is the release that will introduce the deprecation warning).

Reply:
Ah, I understand now. I'll make the change.